### PR TITLE
Standardized power displays

### PR DESCRIPTION
**Pull request by ShizCalev**
🆑 ShizCalev
tweak: All power systems will now report their wattage values in Watts, Kilowatts, Megawatts, and Gigawatts. Gone are the days of seeing 48760 W total load when working with an APC!
/🆑

Units on all objects/interfaces that represent power will now display in the proper measurements. To have an energy number presented in the proper format, you would use the proc DisplayPower(powervarhere), which would then return a text value (along with the appropriate suffix.)

What this means is that instead of showing 48760 Watts on an interface, it will show 48.76 kW (kilowatts.) Likewise for the number 4850000, which would show as 4.85 MW (megawatts), and if you have something crazy that has 1000 MW (or 1 billion Watts), it rolls over into GW (Gigawatts.)

This also helps prevent the issue with very large numbers such as 1.5679e+006 W showing up on interfaces. Instead, the number (which represents 1,567,900 Watts) will be presented instead in it's proper form as 1.568 MW.

Example images;

untitled-1

3nfvqfi

3nfvzdv

a83a4-clip-152kb

### CODE REVIEW

**Comment by MoreRobustThanYou** [comment]
Yessss

**Comment by ShizCalev** [comment]
@kingofkosmos Think this is what you wanted. :)

**Comment by ike709** [comment]
1.21 GW should display as 1.21 JW

**Comment by ShizCalev** [comment]
@ike709 https://en.wikipedia.org/wiki/Watt#Gigawatt

**Comment by ike709** [comment]
@ShizCalev Someone hasn't seen Back to the Future...

**Comment by ShizCalev** [comment]
@Xhuis @KorPhaeron I'm unsure if you'd like cultists to take advantage of this or not, but I checked briefly and it could easily be used by them as well for most of their status displays.

**Comment by PraiseRatvar** [comment]
🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉

**Comment by Xhuis** [comment]
Servants having this might be neat, yeah. I'd say go for it!

**Review by Cyberboss — APPROVED** [review]
Good PR

**Comment by ShizCalev** [comment]
@Xhuis 
3ng2esd

All set.

**Review by Cyberboss — APPROVED** [review]

**Comment by Xhuis** [comment]
#ShizForMaintainer

**Comment by kingofkosmos** [comment]
Wow awesome work @ShizCalev this is so much more than what I wished for.

File: code/game/gamemodes/clock_cult/clock_structure.dm [hunk 1]
```diff
@@ -160,7 +160,7 @@
 	if(is_servant_of_ratvar(user) || isobserver(user))
 		var/powered = total_accessable_power()
 		var/sigil_number = LAZYLEN(check_apc_and_sigils())
-		to_chat(user, "<span class='[powered ? "brass":"alloy"]'>It has access to <b>[powered == INFINITY ? "INFINITY":"[powered]"]W</b> of power, \
+		to_chat(user, "<span class='[powered ? "brass":"alloy"]'>It has access to <b>[powered == INFINITY ? "INFINITE UNITS":"[DisplayPower(powered)]"]</b> of power, \
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structure.dm]: I can't think of a better way to word this than `INFINITE UNITS` but I don't like it.

**Review by ChangelingRain — APPROVED** [review]
delicious. makes clockcult power stuff fit better with the station.

File: code/game/gamemodes/clock_cult/clock_structure.dm (hunk 1 above)
Comment (ShizCalev) [reply code/game/gamemodes/clock_cult/clock_structure.dm]: `UNLIMITED AMOUNTS`?

Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_structure.dm]: "An infinite amount of power", maybe?

Comment (ShizCalev) [reply code/game/gamemodes/clock_cult/clock_structure.dm]: How about endless amount?

Comment (kingofkosmos) [reply code/game/gamemodes/clock_cult/clock_structure.dm]: [`Unlimited power!`](https://youtu.be/Q_7KaMDHoGs?t=183)

File: code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm [hunk 2]
```diff
@@ -193,21 +193,21 @@
 		var/structure_number = 0
 		for(var/obj/structure/destructible/clockwork/powered/P in range(SIGIL_ACCESS_RANGE, src))
 			structure_number++
-		to_chat(user, "<span class='[power_charge ? "brass":"alloy"]'>It is storing <b>[GLOB.ratvar_awakens ? "INFINITY":"[power_charge]"]W</b> of power, \
+		to_chat(user, "<span class='[power_charge ? "brass":"alloy"]'>It is storing <b>[GLOB.ratvar_awakens ? "unlimited</b> amounts":"[DisplayPower(power_charge)]</b>"] of power, \
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm]: maybe `<b>[GLOB.ratvar_awakens ? "INFINITE</b>":"[DisplayPower(power_charge)]</b> of"] power`?

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**